comms/src/builder/builder.rs

```diff
@@ -523,65 +522,30 @@ where
 #[cfg(test)]
 mod test {
     use super::*;
-    use std::path::PathBuf;
-    use tari_storage::lmdb_store::{LMDBBuilder, LMDBError, LMDBStore};
-
-    fn get_path(name: &str) -> String {
-        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-        path.push("tests/data");
-        path.push(name);
-        path.to_str().unwrap().to_string()
-    }
-
-    fn init_datastore(name: &str) -> Result<LMDBStore, LMDBError> {
-        let path = get_path(name);
-        let _ = std::fs::create_dir(&path).unwrap_or_default();
-        LMDBBuilder::new()
-            .set_path(&path)
-            .set_environment_size(10)
-            .set_max_number_of_databases(2)
-            .add_database(name, lmdb_zero::db::CREATE)
-            .build()
-    }
-
-    fn clean_up_datastore(name: &str) {
-        std::fs::remove_dir_all(get_path(name)).unwrap();
-    }
+    use tari_storage::key_val_store::HMapDatabase;
 
     #[test]
     fn new_no_control_service() {
-        let database_name = "builder_new_no_control_service"; // Note: every test should have unique database
-        let datastore = init_datastore(database_name).unwrap();
-        let peer_database = datastore.get_handle(database_name).unwrap();
-
         let comms_services = CommsBuilder::new()
             .with_routes(CommsRoutes::new().register("hello".to_owned()))
             .with_node_identity(NodeIdentity::random_for_test(None))
-            .with_peer_storage(peer_database)
+            .with_peer_storage(HMapDatabase::new())
```

Comment:
Currently HMapDatabase is only used in Comms but could be used everywhere the peer_database is used during testing. This will unfortunately require that PeerManager receive a generic for specifying the key-val store trait and each struct that use a PeerManager will also need an extra generic.  This seems slightly ugly, but might be worth it as no persistent storage files will be created during testing.